core/org.wso2.carbon.bootstrap/src/test/java/org/wso2/carbon/bootstrap/BootstrapTest.java

```diff
@@ -0,0 +1,139 @@
+ * limitations under the License.
+ */
+package org.wso2.carbon.bootstrap;
+
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.file.Files;
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertTrue;
+
+/**
+ * Tests functionality of Bootstrap class
+ */
+public class BootstrapTest {
+    private Bootstrap bootstrap = new Bootstrap();
+    private final Set<URL> classpath = new LinkedHashSet<URL>();
+    private File libFile;
+
+    /**
+     * Create necessary temporary files.
+     *
+     * @throws IOException for file operation failures
+     */
+    @BeforeMethod
+    public void setUp() throws IOException {
+        libFile = Files.createTempDirectory("libFile").toFile();
+    }
+
+    /**
+     * Test if loadClass throws ClassNotFoundException when trying to load
+     * a class that does not exist within the scope to load the rest of the app
+     *
+     * @throws Exception for failure of file operations or classloading
+     */
+    @Test(expectedExceptions = ClassNotFoundException.class)
+    public void testLoadClassException() throws Exception {
+        bootstrap.addFileUrl(libFile);
+        bootstrap.addClassPathEntries();
+        ClassLoader cl = new URLClassLoader(classpath.toArray(new URL[classpath.size()]));
+        assertNotNull(cl);
+        cl.loadClass("simple.test.class");
+    }
+
+    /**
+     * Test if addClassPathEntries method sets a given directory path as the
+     * lib path
+     *
+     * @throws Exception for malformed URL or file operation failures
+     */
+    @Test
+    public void testAddClassPathEntries() throws Exception {
+        System.setProperty("carbon.internal.lib.dir.path", libFile.getParent());
+        bootstrap.addFileUrl(libFile);
+        bootstrap.addClassPathEntries();
+    }
+
+    /**
+     * Test if addFileUrl method throws MalformedURLException for unknown protocols
+     *
+     * @throws MalformedURLException if a URL is malformed
+     */
+    @Test(expectedExceptions = MalformedURLException.class)
+    public void testAddFileUrl() throws MalformedURLException {
+        URL obj = new URL("fil://path/to/file");
+        File libFile = new File(obj.toString());
+        bootstrap.addFileUrl(libFile);
+    }
+
+    /**
+     * Test if addJarFileUrls method throws malformedURLException for unknown protocols
+     *
+     * @throws MalformedURLException if a URL is malformed
+     */
+    @Test(expectedExceptions = MalformedURLException.class)
+    public void testAddJarFileUrls() throws MalformedURLException {
+        URL obj = new URL("fil://path/to/file");
+        File libFile = new File(obj.toString());
+        bootstrap.addFileUrl(libFile);
```

Comment:
Don't we need to invoke the 'addJarFileUrls' method for this test case?

Reply:
Fixed in f60a4aa

Reply:
Fixed in 527e06d